### ISSUE TITLE
Pin on main Webpage 

### ISSUE THREAD

**Issue by hadphild** [bitfocus/companion#573]
**Describe the feature**
Set a password for companion webpage

**Is this platform dependent (windows, mac, ..)?**
All

**Usecases**
Protecting your companion from others that could login to your config. Be good for installs and not letting other play with your config.

**Comment by mschoeffmann** [comment]
As mentioned in #103, I think a protection against unauthorized changes would be important for fixed installations.

I'm only talking about AdminGUI now, all other "trigger" protocols (osc, http, artnet, ...) can remain unprotected since control networks are normally closed (further, these protocols can only be protected by IP filter because not all of them support authentication functions).

Having this accessible without protection ("user mode"):
- Startpage
- Logs
- Enable/Disable (restart) connections/instances
- Web-Buttons
- Emulator (respecting the PIN code settings like now)

And this after login to "admin mode":
- Buttons
- Surfaces
- Settings
- Presets
- Edit connections/instances
- Import/Export of configuration

**Comment by stevespaw** [comment]
Yes we could use this for fixed installs also on the admin GUI. If they are running the emulator.
Thanks

**Comment by Julusian** [comment]
In 2.2 there is a fairly insecure admin ui lock system in place.
It is not designed to keep someone determined out (any knowledge of javascript and it can be bypassed).

Leaving this open for a proper authentication system to be implemented
